src/util/alloc/allocator.rs

```diff
@@ -318,6 +318,19 @@ pub trait Allocator<VM: VMBinding>: Downcast {
     /// * `offset` the required offset in bytes.
     fn alloc_slow_once(&mut self, size: usize, align: usize, offset: usize) -> Address;
 
+    /// A wrapper method for [`alloc_slow_one`](Allocator::alloc_slow_once) to insert USDT tracepoints.
+    ///
+    /// Arguments:
+    /// * `size`: the allocation size in bytes.
+    /// * `align`: the required alignment in bytes.
+    /// * `offset` the required offset in bytes.
+    fn alloc_slow_once_traced(&mut self, size: usize, align: usize, offset: usize) -> Address {
+        probe!(mmtk, alloc_slow_once_start);
```

Comment:
These trace points are pretty clear. You can use the same pattern for gc start/end, and work start/end.

Reply:
Done